src/entities/Matching.ts

```diff
@@ -42,40 +36,18 @@ class Matching {
   users: User[];
 
   serialize(): SerializedMatching {
-    const callbackSchedule = (
-      accum: SerializedDaySchedule[],
-      currentVal: DaySchedule
-    ) => {
-      accum.push(currentVal.serialize());
-      return accum;
-    };
-    const callbackUser = (accum: SerializedUser[], currentVal: User) => {
-      accum.push(currentVal.serialize());
-      return accum;
-    };
     return {
       active: this.active,
-      schedule: this.schedule.reduce(callbackSchedule, []),
-      users: this.users.reduce(callbackUser, []),
+      schedule: this.schedule.map(schedule => schedule.serialize()),
+      users: this.users.map(user => user.serialize()),
     };
   }
 
   subSerialize(): SubSerializedMatching {
-    const callbackSchedule = (
-      accum: SerializedDaySchedule[],
-      currentVal: DaySchedule
-    ) => {
-      accum.push(currentVal.serialize());
-      return accum;
-    };
-    const callbackUser = (accum: SubSerializedUser[], currentVal: User) => {
-      accum.push(currentVal.subSerialize());
-      return accum;
-    };
     return {
       active: this.active,
-      schedule: this.schedule.reduce(callbackSchedule, []),
-      users: this.users.reduce(callbackUser, []),
+      schedule: this.schedule.map(schedule => schedule.serialize()),
```

Comment:
wait.. is `this.schedule` an array? cuz you can only `map` an array right.. why is it not called `this.schedules`? i might just also be dumb

Reply:
Lol, it's an array because it's of type `daySchedule[]` as a schedule consists of multiple day schedules. It's kinda just a weird english thing haha.